src/harness/testcases/WINNF_FT_S_MCP_testcase.py

```diff
@@ -0,0 +1,243 @@
+    # Remove conditionals from registration
+    del device_2['cbsdCategory']
+    del device_2['airInterface']
+    del device_2['installationParam']
+    del device_2['measCapability']
+   
+    # Load GWPZ Record
+    gwpz_record_1 = json.load(
+      open(os.path.join('testcases', 'testdata', 'gwpz_record_0.json')))
+
+    # Load FSS record
+    fss_record_1 = json.load(
+      open(os.path.join('testcases', 'testdata', 'fss_record_0.json')))
+
+    # Load ESC record
+    esc_record_1 = json.load(
+      open(os.path.join('testcases', 'testdata', 'esc_sensor_record_0.json')))
+  
+    # Load PPA record
+    ppa_record = json.load(
+      open(os.path.join('testcases', 'testdata', 'ppa_record_0.json'))) 
+    pal_record = json.load(
+      open(os.path.join('testcases', 'testdata', 'pal_record_0.json')))
+
+    pal_low_frequency = 3550000000.0
+    pal_high_frequency = 3560000000.0        
+
+    ppa_record_1, pal_record_1 = makePpaAndPalRecordsConsistent(ppa_record,
+                                                                [pal_record],
+                                                                pal_low_frequency,
+                                                                pal_high_frequency,
+                                                                'test_user_1')
+    # Load DPA record
+    dpa_1 = json.load(
+      open(os.path.join('testcases', 'testdata', 'dpa_0.json')))
+    dpa_2 = json.load(
+      open(os.path.join('testcases', 'testdata', 'dpa_1.json')))
+    dpa_3 = json.load(
+      open(os.path.join('testcases', 'testdata', 'dpa_0.json')))
+
+    # Registration and grant records for multiple iterations
+    cbsd_records_iter_1_domain_proxy_0 = {
+        'registrationRequests': [device_1, device_2],
+        'grantRequests': [grant_request_1, grant_request_2],
+        'conditionalRegistrationData': conditionals
+    }
+    cbsd_records_iter_1_domain_proxy_1 = {
+        'registrationRequests': [device_3],
+        'grantRequests': [grant_request_3],
+        'conditionalRegistrationData': {}
+    }
+    cbsd_records_iter_2_domain_proxy_0 = {
+        'registrationRequests': [device_4],
+        'grantRequests': [grant_request_4],
+        'conditionalRegistrationData': {}
+    }
+    cbsd_records_iter_2_domain_proxy_1 = {
+        'registrationRequests': [device_5, device_6],
+        'grantRequests': [grant_request_5, grant_request_6],
+        'conditionalRegistrationData': {}
+    }
+    
+    # Protected entities records for multiple iterations
+    protected_entities_iter_1 = {
+        'palRecords': [pal_record_1],
+        'ppaRecords': [ppa_record_1], 
+        'escRecords': [esc_record_1]
+    }
+    protected_entities_iter_2 = {
+        'gwpzRecords': [gwpz_record_1], 
+        'fssRecords': [fss_record_1]
+    }
+
+    # SAS Test Harnesses configurations, 
+    # Following configurations are for two SAS test harnesses for two iterations
+    sas_test_harness_device_1 = json.load(
```

Comment:
You need to be careful to not reuse the same (FCC ID, SN) combinations -- otherwise the SAS will reject (this logic is tested in GRA.5,6, which you are also writing).